examples/tree_1d_dgsem/elixir_shallowwater_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,183 @@
+
+  if x[1] >= 0.5
+    b = sin(4 * pi * x[1]) + 1
+  end
+
+  H = max(b, 2.5)
+
+  if x[1] >= 0.5
+    H = max(b, 1.5)
+  end
+
+  # It is mandatory to shift the water level at dry areas to make sure the water height h
+  # stays positive. The system would not be stable for h set to a hard 0 due to division by h in
+  # the computation of velocity, e.g., (h v) / h. Therefore, a small dry state threshold
+  # (1e-13 per default, set in the constructor for the ShallowWaterEquations) is added if h = 0.
+  # This default value can be changed within the constructor call depending on the simulation setup.
+  H = max(H, b + equations.threshold_limiter)
+  return prim2cons(SVector(H, v, b), equations)
+end
+
+initial_condition = initial_condition_well_balanced_chen_noelle
+
+###############################################################################
+# Get the DG approximation space
+
+volume_flux = (flux_wintermeyer_etal, flux_nonconservative_wintermeyer_etal)
+surface_flux = (FluxHydrostaticReconstruction(flux_hll_chen_noelle, hydrostatic_reconstruction_chen_noelle),
+                flux_nonconservative_chen_noelle)
+
+basis = LobattoLegendreBasis(3)
+
+indicator_sc = IndicatorHennemannGassner(equations, basis,
+                                         alpha_max=0.5,
+                                         alpha_min=0.001,
+                                         alpha_smooth=true,
+                                         variable=waterheight_pressure)
+volume_integral = VolumeIntegralShockCapturingHG(indicator_sc;
+                                                 volume_flux_dg=volume_flux,
+                                                 volume_flux_fv=surface_flux)
+
+solver = DGSEM(basis, surface_flux, volume_integral)
+             
+###############################################################################
+# Create the TreeMesh for the domain [0, 1]
+
+coordinates_min = 0.0
+coordinates_max = 1.0
+
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level=6,
+                n_cells_max=10_000)
+
+# create the semi discretization object
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 25.0)
+ode = semidiscretize(semi, tspan)
+
+function initial_condition_discontinuous_well_balancedness(x, t, element_id, equations:: ShallowWaterEquations1D)
+  v = 0.0
+  b = sin(4 * pi * x[1]) + 1
+  H = max(b, 1.5)
+
+  # for refinement_level=6
+  if (element_id >= 1 && element_id <= 32)
+    b = sin(4 * pi * x[1]) + 3
+    H = max(b, 2.5)
+  end
+
+  # Shift the water level by the amount `equations.threshold_limiter` (1e-13 per default)
+  # to avoid division by a hard 0 value in the water height `h` when computing velocities.
+  H = max(H, b + equations.threshold_limiter)
+  return prim2cons(SVector(H, v, b), equations)
+end
+
```

Comment:
This can be simplified with a similar strategy used in the 2D version of this elixir. Then we only need one version of the initial condition routine and we can set the discontinuity using the same perturbed node strategy.